src/uShip.Logging.Tests/FluentLoggerDataBuilderTestscs.cs

```diff
@@ -0,0 +1,121 @@
+            Assert.AreEqual(ClassUnderTest._data[objName + "LongProp"], testLong);
+            Assert.AreEqual(ClassUnderTest._data[objName + "DoubleProp"], testDouble);
+            Assert.AreEqual(ClassUnderTest._data[objName + "FloatProp"], testFloat);
+            ClassUnderTest._data.Count.Should().Be(5);
+        }
+
+        [Test]
+        public void Can_map_object_with_complex_types()
+        {
+            string testString = "test";
+            int testInt = 32;
+            long testLong = 64L;
+            double testDouble = (double)2.0;
+            float testFloat = (float)3.0;
+
+            var obj = new
+            {
+                StringProp = testString,
+                IntegerProp = testInt,
+                LongProp = testLong,
+                DoubleProp = testDouble,
+                FloatProp = testFloat,
+                ComplexClass = new {
+                    NestedProperty = "Oh no!"
+                }
+            };
+            ClassUnderTest.Data(obj);
+            var objName = obj.GetType().Name + "_";
+            Assert.AreEqual(ClassUnderTest._data[objName + "StringProp"], testString);
+            Assert.AreEqual(ClassUnderTest._data[objName + "IntegerProp"], testInt);
+            Assert.AreEqual(ClassUnderTest._data[objName + "LongProp"], testLong);
+            Assert.AreEqual(ClassUnderTest._data[objName + "DoubleProp"], testDouble);
+            Assert.AreEqual(ClassUnderTest._data[objName + "FloatProp"], testFloat);
+            ClassUnderTest._data.Count.Should().Be(5);
+            ClassUnderTest._data.ContainsKey(objName + "ComplexClass").Should().Be(false);
+            ClassUnderTest._data.ContainsKey(objName + "ComplexClass_NestedProperty").Should().Be(false);
+        }
+
+        [Test]
+        public void Can_map_object_with_dates()
+        {
+            string testString = "test";
+            int testInt = 32;
+            long testLong = 64L;
+            double testDouble = (double)2.0;
+            float testFloat = (float)3.0;
+            DateTime testDate = new DateTime(66, 6, 6);
+            DateTimeOffset testDateOffset = new DateTimeOffset(testDate, TimeSpan.FromHours(2));
+
+            var obj = new
+            {
+                StringProp = testString,
+                IntegerProp = testInt,
+                LongProp = testLong,
+                DoubleProp = testDouble,
+                FloatProp = testFloat,
+                ComplexClass = new
+                {
+                    NestedProperty = "Oh no!"
+                },
+                DateTimeProp = testDate,
+                DateTimeOffsetProp = testDateOffset
+            };
+            ClassUnderTest.Data(obj);
+            var objName = obj.GetType().Name + "_";
+            Assert.AreEqual(ClassUnderTest._data[objName + "StringProp"], testString);
+            Assert.AreEqual(ClassUnderTest._data[objName + "IntegerProp"], testInt);
```

Comment:
Again, I think it would be best to keep the test focused on just one thing at a time.  Repeating the primitive test here makes all of these tests somewhat interdependent.